lib/rouge/lexers/magik.rb

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Magik < RegexLexer
+      title "Magik"
+      desc "Smallworld Magik"
+      tag 'magik'
+      aliases 'magik'
+      filenames '*.magik'
+      mimetypes 'text/x-magik', 'application/x-magik'
+
+      keywords = %w(
+           _package
+           _pragma
+           _block _endblock
+           _handling _default
+           _protect _protection _endprotect
+           _try _with _when _endtry
+           _catch _endcatch
+           _throw
+           _lock _endlock
+           _if _then _elif _else _endif
+           _for _over _loop _endloop _loopbody _continue _leave
+           _return
+           _local _constant _recursive _global _dynamic _import
+           _private _iter _abstract _method _endmethod
+           _proc _endproc
+           _gather _scatter _allresults _optional
+           _thisthread _self _clone _super
+           _primitive
+           _unset _true _false _maybe
+           _is _isnt _not _and _or _xor _cf _andif _orif
+           _div _mod
+        )
+
+      string_double = /"[^"\n]*"/
+      string_single = /'[^'\n]*'/
+
+      digits = /[0-9]+/
+      hex_digits = /[0-9a-f]+/i
+      radix = /r#{hex_digits}/i
```

Comment:
Can hexadecimal digits appear outside of a radix?

Reply:
Fixed.